### PR TITLE
fix helm init --upgrade logic

### LINKED ISSUE #3392: `helm --upgrade` "error when upgrading: current Tiller version is newer" when current and new version are identical
# Description

`helm init --upgrade`  
seems to think `v2.8.0` is greater than `v2.8.0`

Using this version:
```sh
$ helm version
Client: &version.Version{SemVer:"v2.8.0", GitCommit:"14af25f1de6832228539259b821949d20069a222", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.8.0", GitCommit:"14af25f1de6832228539259b821949d20069a222", GitTreeState:"clean"}
```

I get this error:
```sh
$ helm init --upgrade
$HELM_HOME has been configured at /Users/redacted/.helm.
Error: error when upgrading: current Tiller version is newer, use --force-upgrade to downgrade

$ helm init --force-upgrade # force-upgrade is fine
$HELM_HOME has been configured at /Users/redacted/.helm.
Warning: Tiller is already installed in the cluster.
(Use --client-only to suppress this message, or --upgrade to upgrade Tiller to the current version.)
Happy Helming!

$ helm init --upgrade # still an issue after force upgrade
$HELM_HOME has been configured at /Users/redacted/.helm.
Error: error when upgrading: current Tiller version is newer, use --force-upgrade to downgrade
```

# Expected:

`helm init --upgrade` 

should work if the current version and the new version are identical.

#

### PR DESCRIPTION
**Pull request by bacongobbler**
This change does the following:

- performs an upgrade anytime we're using non-semver-compatible clients or servers (e.g. `helm init --canary-image && helm init --upgrade`, or just by installing a canary client). Beforehand, it would panic when it hit `semver.MustParse`.
- allows upgrades when the clients are identical, as CI systems relied on this to not error out or as a no-op. The system will allow upgrades when the client and server are identical to allow the user to tweak settings

downgrades are still not allowed without the use of `--force-upgrade`.

closes #3392

### CODE REVIEW

**Review by thomastaylor312 — APPROVED** [review]
This looks well tested!

**Comment by thomastaylor312** [comment]
This is labeled as a large, so we should probably get a second review

**Review by SlickNik — APPROVED** [review]
Looks good to me! Thanks @bacongobbler - merge when ready. 👍

**Pull request merged**
